### DISCUSSION TITLE
Como obter os casos activos por ARS (ou concelho)

### DISCUSSION THREAD

**Discussion by tpaixao** [dssg-pt/covid19pt-data#508] [Dados Nacionais e por ARS]
Descontando o numero de recuperados e obitos dos casos confirmados por ARS não dá nem de perto os mesmo valores do numero global de casos activos. Será uma falha da DGS em reportar os recuperados e/ou Obitos? Há alguma maneira de obter esta informação?

obrigado!

**Comment by davipt** [comment]
(correção… não tinha visto "por ARS" 😬 sorry)

A soma de confirmados por ARS comparado com os confirmados nacionais teve uma diferença de 1 caso a 16-03-2020 e de 3 a 20-03-2020. São dois dias em que não se conseguiu descobrir qual das regiões tem o valor incorrecto. Mas no dia seguinte está corrigido. A soma de óbitos por ARS comparado com os óbitos nacionais apenas teve uma diferença de 10 casos a 24-10-2020. 🤔

A diferença de confirmados totais subtraindo recuperados e subtraindo óbitos dá o valor de activos. Este repositório tem testes automáticos para validar este valor e outros antes de ser aprovado e publicado.

Como está a fazer as contas para podermos tentar reproduzir? Estarás a olhar para o campo de recuperados por região, que só teve valores durante poucos dias no inicio da pandemia?

Confirmei os dados para 23-01-2021:
- A soma de confirmados por ARS é 280156 + 87095 + 214830 + 21372 + 14641 + 3180 + 3195 = 624469, igual ao total de confirmados.
- A soma de óbitos por ARS é 4020 + 1711 + 3748 + 505 + 157 + 22 + 31 = 10194, igual ao total de óbitos.
- A diferença de confirmados 624469  menos óbitos 10194 menos recuperados 451324 (que não existe por ARS) dá o valor de activos 162951.

**Reply by tpaixao** [reply]
eu estou a fazer literalmente isto:

```
for reg in ['arslvt','arscentro','arsalentejo','arsalgarve','arsnorte','acores','madeira']:
    df_data['activos_'+reg]=df_data['confirmados_'+reg]-df_data['recuperados_'+reg]-df_data['obitos_'+reg]
```

Depois sumando essas colunas de activos por regiao (ars + regioes autonomas):
```
sum([df_data['activos_'+reg] for reg in ['arslvt','arscentro','arsalentejo','arsalgarve','arsnorte','acores','madeira'] ])
```
não dá os mesmo numeros que a coluna `ativos` ... O que estou a fazer de errado?

**Reply by davipt** [reply]
df_data['recuperados_'+reg] 😉

Os valores de recuperados por região só foram publicados entre 23-03-2020 e 01-04-2020.
O resto está a zeros, portanto a diferença entre a tua soma e a coluna activos deverá ser quase o total de recuperados.

Sem estes dados não dá para calcular os activos por região.  Percebo o que querias fazer, mas não é possível. Talvez desse para derivar a partir das relações percentuais entre os dados existentes de confirmados e óbitos, mas ainda assim seria apenas aproximado. Recuperados e Activos são os valores menos relevantes pois evoluem mais por decisões de secretaria (2 vs 1 teste negativo, e 14 vs 10 dias de recuperação) do que por variação real da pandemia.

Faz sentido?

**Comment by tpaixao** [comment]
Obrigado, faz todo o sentido. nao me apercebi que esses dados só tinham sido publicados nessas datas. Verificar sempre os dados!
